crates/burn-jit/src/fusion/on_write/builder.rs

```diff
@@ -28,6 +58,13 @@ impl OptimizationBuilder<FuseOnWriteTrace> for FuseOnWriteBuilder {
             return;
         }
 
+        // TODO: If tensors are re-used, we should count them only once.
+        self.num_tensors += op.nodes().len();
+        if self.num_tensors >= 10 {
+            self.status = OptimizationStatus::Closed;
+            return;
+        }
```

Comment:
It should use `estimate_bindings` I think, and 10 should be replaced by the value in the device's hardware properties.

Reply:
Woops this was leftover dummy stuff